vue/components/organisms/configurator/configurator.vue

```diff
@@ -507,19 +492,36 @@ export const Configurator = {
                 }
             });
 
-            this.$bus.bind("highlight_part", part => {
-                if (this.ignoreBus) return;
-                if (this.configurator && this.configurator.ready) {
-                    this.configurator.highlight(part);
-                }
-            });
+            if (this.configuratorType === "prc") {
+                this.configurator.bind("highlighted_part", part => {
+                    this.$bus.trigger("highlighted_part", this.configurator, part);
+                });
 
-            this.$bus.bind("lowlight_part", part => {
-                if (this.ignoreBus) return;
-                if (this.configurator && this.configurator.ready) {
-                    this.configurator.lowlight(part);
-                }
-            });
+                this.configurator.bind("lowlighted", () => {
+                    this.$bus.trigger("lowlighted", this.configurator);
+                });
+
+                this.$bus.bind("highlight_part", part => {
+                    if (this.ignoreBus) return;
+                    if (this.configurator && this.configurator.ready) {
+                        this.configurator.highlight(part);
+                    }
+                });
+
+                this.$bus.bind("lowlight_part", part => {
+                    if (this.ignoreBus) return;
+                    if (this.configurator && this.configurator.ready) {
+                        this.configurator.lowlight(part);
+                    }
+                });
+            } else if (this.configuratorType === "csr") {
+                this.configurator.bind("ready", options => {
+                    if (options?.origin?.startsWith("configurator")) {
+                        this.loading = false;
+                        this.configurator.resize();
+                    }
+                });
+            }

```

Comment:
Separate these binds into two different methods like `prcBinds()` and `csrBinds()` or something like that